state-chain/pallets/cf-elections/src/vote_storage/composite.rs

```diff
@@ -232,3 +232,41 @@ generate_vote_storage_tuple_impls!(tuple_1_impls: (A));
 generate_vote_storage_tuple_impls!(tuple_2_impls: (A, B));
 generate_vote_storage_tuple_impls!(tuple_3_impls: (A, B, C));
 generate_vote_storage_tuple_impls!(tuple_4_impls: (A, B, C, D));
+
+use crate::electoral_systems::blockchain::delta_based_ingress::{
+	ChannelTotalIngressed, MAXIMUM_CHANNELS_PER_ELECTION,
+};
+/// Implementation of BenchmarkValue for the concrete types used in our Runtime.
+/// Needed for Solana Witnessing.
+use cf_chains::{
+	benchmarking_value::BenchmarkValue,
+	sol::{SolAddress, SolHash},
+	Solana,
+};
+use frame_support::BoundedBTreeMap;
+use sp_core::ConstU32;
+use tuple_4_impls::{CompositeSharedData, CompositeVote};
+impl BenchmarkValue
+	for CompositeVote<
+		u64,
+		u64,
+		BoundedBTreeMap<
+			SolAddress,
+			ChannelTotalIngressed<Solana>,
+			ConstU32<MAXIMUM_CHANNELS_PER_ELECTION>,
+		>,
+		SolHash,
+	>
+{
+	#[cfg(feature = "runtime-benchmarks")]
+	fn benchmark_value() -> Self {
+		CompositeVote::A(1_000u64)
+	}
+}
```

Comment:
Can we not implement this generically? 

```suggestion
impl<A, B, C, D> BenchmarkValue for CompositeVote<A, B, C, D>
	where A: BenchmarkValue,
{
	#[cfg(feature = "runtime-benchmarks")]
	fn benchmark_value() -> Self {
		CompositeVote::A(A::benchmark_value())
	}
}
```

And the same for SharedData ?